### ISSUE TITLE
Some unordered lists to convert to definition lists

### ISSUE THREAD

**Issue by nschonni** [mdn/content#537]
There are some regular lists that appears that they should probably get converted over to definition lists
https://github.com/mdn/content/search?q=%3Cli%3E%3Cstrong%3E
Not sure if this is easy to be automated though, so I thought an issue would be good before trying anything

**Comment by mattiapontonio** [comment]
FTR
https://stackoverflow.com/q/38288398/3579536

**Comment by Ryuno-Ki** [comment]
Dang! I can't find the article, but please check `<dl>`s on accessible technology before changing the markup.
I recall a discussion on Twitter, where it was around FAQs (IIRC). The lesson learned was, that `<dl>` have spotty support on AT.

**Comment by mattiapontonio** [comment]
In general i don't like things like that:

This is a list:
- This is the key**:** this is the value
- I'm good because I'm something in a list and not something in something in a list

**Comment by Ryuno-Ki** [comment]
I agree with you, that it would make semantically more sense to use `<dl>`s. However, if the support is spotty on the User Agent side … Perhaps not the best idea?
I'm asking for some research and/or testing before making changes.

**Comment by mattiapontonio** [comment]
> I agree with you, that it would make semantically more sense to use `<dl>`s. However, if the support is spotty on the User Agent side … Perhaps not the best idea?
> I'm asking for some research and/or testing before making changes.

There are **230** code results.
Is possibile to automate not the Regex replace in 1 PR with 230 file changhes but 230 issues with potential 230 PRs?

**Comment by mattiapontonio** [comment]
> Dang! I can't find the article, but please check `<dl>`s on accessible technology before changing the markup.
> I recall a discussion on Twitter, where it was around FAQs (IIRC). The lesson learned was, that `<dl>` have spotty support on AT.

Here
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dl

The BCT table Is all `green` so seems ok.
Where is the _AT-BCT (AT Browser compatibility table)_ or the source to assert this support issue?

**Comment by Ryuno-Ki** [comment]
Finally found it!

https://adrianroselli.com/2020/09/voiceover-on-ios-14-supports-description-lists.html by @aardrian

Or see https://twitter.com/scottohara/status/1256998880593555464 by @scottaohara

So basically, it _can_ work out, but is not guaranteed.

**Comment by Ryuno-Ki** [comment]
I just remembered a conversation with Adrian I had on Twitter a while ago, where he introduced me to a collection of screenreader tests. See https://a11ysupport.io/tech/html/dl_element for dl.

**Comment by mattiapontonio** [comment]
Thanks, but I didn't understand in this case we should add some aria attributes to our Regex ...?

**Comment by mattiapontonio** [comment]
However, I was thinking that a Regex is not enough. We need to create a script that matches the lists and performs a new regex on all the matches which then replaces the items. It is an indented Regex.

**Comment by mattiapontonio** [comment]
@Ryuno-Ki can You review this?
https://github.com/mdn/content/pull/729

It's 1 of X of these lists.
I don't know now How to request review in the fancy GitHub way etc...

**Comment by mattiapontonio** [comment]
Note for the title:

Here are <dl>: The **Description** List element

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dl

**Comment by Ryuno-Ki** [comment]
> @Ryuno-Ki can You review this?
> #729

I can. Will do so after dinner (in about two hours from now).

**Comment by sideshowbarker** [comment]
To take this back to the original issue description:

> There are some regular lists that appears that they should probably get converted over to definition lists
> [mdn/content/search (* **)**](https://github.com/mdn/content/search?q=%3Cli%3E%3Cstrong%3E)
> Not sure if this is easy to be automated though, so I thought an issue would be good before trying anything

Some pattern like `<li><strong>.+</strong>\s?[-:,]` would seem to be more precise/effective for identifying such cases.

I ran `rg "<li><strong>.+</strong>\s?[-:,]" files/en-us/web` locally and found quite a lot of cases of normal lists that *could* be expressed as definition lists instead. For example, `files/en-us/web/web_components/using_shadow_dom/index.html` has:

```html
<p>There are some bits of shadow DOM terminology to be aware of:</p>

<ul>
 <li><strong>Shadow host</strong>: The regular DOM node that the shadow DOM is attached to.</li>
 <li><strong>Shadow tree</strong>: The DOM tree inside the shadow DOM.</li>
 <li><strong>Shadow boundary</strong>: the place where the shadow DOM ends, and the regular DOM begins.</li>
 <li><strong>Shadow root</strong>: The root node of the shadow tree.</li>
</ul>
```
But I’d argue that list is just fine as-is. The main effect that turning it into a definition list would seem to have is just to add more vertical space between the items.

That said, however, I did also find around 25 or so articles with stuff shoehorned into normal lists that would clearly be improved by using definition lists instead. For an example of one particularly extreme case, take a look at the lists in https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/preserveAspectRatio#syntax — and also take a look at the markup of those lists in the source, too. It’s really a big mess.

So if somebody wanted to take the time to put together a patch to re-work those particular messes into `dl` lists, I guess that is one case where I personally at least would see that as a real improvement and would support merging it. However, given that part of what was surfaced in https://github.com/mdn/content/issues/4367 is that representing definition lists in markdown is really problematic, I’m not sure how keen in general others would be at this point on merging changes that replaced any normal lists with definition lists.

In other words, given that we’re moving to markdown and that representing definition lists in markdown is a bit of a complication that it seems we want to try to avoid, it therefore seems unlikely any of us ourselves are going to end up making this issue a priority. Certainly I can say that I definitely couldn’t justify putting time into it myself. And it doesn’t seem like a good candidate for labeling with “help wanted” to encourage other contributors to work on it.

So I think we possibly should consider closing this issue — unless I’m wrong about it all and others disagree and instead think we really should make this a priority and/or add a “help wanted” label to it.

**Comment by Ryuno-Ki** [comment]
> I can. Will do so after dinner (in about two hours from now).

No, I not starved. I simply forgot about this issue :facepalm:

**Comment by wbamberg** [comment]
I'm closing this as it doesn't seem like a clear enough statement of the problem for us to work on it. If specific pages are identified that should be `<dl>` elements we can fix them.
